cookbook/ui_customization/add_custom_informations_to_a_field.rst

```diff
@@ -0,0 +1,91 @@
+How to add custom informations to a field
+=========================================
+
+Natively we add a lot of related informations to product edit fields: data coming from a variant group, validation errors, etc. As an integrator, you can also add custom informations for your own needs and we will go through each steps in this cookbook to acheive this.
+
+Let's say that we want to display the minimum and maximum values allowed for a number field, for that we need to create a form extension listening to the `pim_enrich:form:field:extension:add` event:
+
+.. code-block:: javascript
+    :linenos:
+
+    'use strict';
+    /**
+     * src/Acme/Bundle/CustomBundle/Resources/public/js/product/form/attributes/number-min-max.js
+     */
+    define(
+        [
+            'jquery',
+            'underscore',
+            'pim/form'
+        ],
+        function ($, _, BaseForm) {
+            return BaseForm.extend({
+                configure: function () {
+                    this.listenTo(this.getRoot(), 'pim_enrich:form:field:extension:add', this.addFieldExtension);
+
+                    return BaseForm.prototype.configure.apply(this, arguments);
+                },
+                addFieldExtension: function (event) {
+                    //The event contains the field and an array of promises. The field will wait for all the promises to be resolve before rendering.
+                    //You can add a promise to this array to be sure that the field will wait for you before rendering itself
+                    event.promises.push($.Deferred().resolve().then(function () {
```

Comment:
I think it's not very readable for a beginner js dev. Maybe using the regular `var deferred = $.Deferred()` syntax would be easier to understand (just here).